shared/main.tf

```diff
@@ -0,0 +1,73 @@
+provider "azurerm" {
+    version = "~>1.21.0"
+}
+
+terraform {
+  required_version = "~> 0.11.11"
+  backend "azurerm" {
+  }
+}
+
+locals {
```

Comment:
for readability can you indent this?

Reply:
Done automatically with merge from master.